lib/exercism/user.rb

```diff
@@ -52,21 +52,21 @@ def access?(problem)
   end
 
   # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
-  def self.from_github(id, username, email, avatar_url, joined_as=nil)
-    user = User.where(github_id: id).first
+  def self.from_authentication_data(github_user, joined_as=nil)
```

Comment:
Why have you renamed this method?
Is it now any less dependent on GitHub?
Why isn't joined_as part of `github_user`?
(I really like the collection of related parameter data into an object.)

This is another method that does many things, how many are there? How many of these responsibilities could be extracted? How many of them really belong in the User model?